src/core/qgsdataitem.h

```diff
@@ -143,6 +143,11 @@ class CORE_EXPORT QgsDataItem : public QObject
      */
     virtual QList<QAction *> actions() { return QList<QAction *>(); }
 
+    /** Returns the list of menus available for this item. This is usually used for the popup menu on right-clicking
+     * the item. Subclasses should override this to provide actions.
```

Comment:
Can you add a note here:

"Ownership of the returned menus is unchanged. Implementations should take care to ensure that created menus are correctly parented (e.g. by setting the parent to the data item) to avoid leaking the returned menus."